tests/test_builders.py

```diff
@@ -90,6 +92,140 @@ def test_build_protein_system_returns_correct_water_count():
             last_num_waters = host_config.num_water_atoms
 
 
+@pytest.mark.nocuda
+@pytest.mark.parametrize("ionic_concentration", [0.0, 0.15])
+@pytest.mark.parametrize("neutralize", [False, True])
+def test_water_system_ion_concentration_and_neutralization(ionic_concentration, neutralize):
+    positive_mol = ligand_from_smiles("c1cc[nH+]cc1")
+    negative_mol = ligand_from_smiles("[N+](=O)([O-])[O-]")
+    neutral_mol = ligand_from_smiles("c1ccccc1")
+
+    box_size = 2.0
+
+    host_config_no_ions = build_water_system(box_size, DEFAULT_WATER_FF, ionic_concentration=0.0)
+    # Host system will have zero net charge if no ionic concentration and not neutralized
+    assert np.sum(host_config_no_ions.host_system.nonbonded_all_pairs.params[:, NBParamIdx.Q_IDX]) == 0.0
+
+    # Can't mix ligands of different charges when neutralizing the system
```

Comment:
Is there a hard reason for this? (Eg. adding a -1 mol and +1 mol should end up in not needing any neutralization)

Reply:
ah got it, so typical use case is `len(mols)=2` that correspond to the end-state molecules. I think later on, after we refactor the single topology code, we'll be able to seamlessly morph counter-ions into waters or other counter-ions to neutralize the system.